cyberdrop_dl/utils/dataclasses/url_objects.py

```diff
@@ -37,6 +37,8 @@ def __init__(self, url: "URL", parent_title: str, part_of_album: bool = False, a
                  retry: bool = False, retry_path: Path = None):
         self.url: URL = url
         self.parent_title: str = parent_title
+        # WARNING: unsafe but deepcopy is used when a new child item is created
+        self.parents : list[URL] = []
```

Comment:
What makes this unsafe?

Reply:
Every new `ScrapeItem` will point to the same list in memory. Changing the parents of one will affect all of them. However, all the crawlers use one single root `ScrapeItem` and new ones are actually a `deepcopy` of the previous one, so it should be fine.

The only crawler that didn't do this was `Fapello` but I just pushed a commit to fix it as well.